tachyon/crypto/commitments/kzg/kzg.h

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_CRYPTO_COMMITMENTS_KZG_KZG_H_
+#define TACHYON_CRYPTO_COMMITMENTS_KZG_KZG_H_
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/math/elliptic_curves/msm/variable_base_msm.h"
+#include "tachyon/math/elliptic_curves/point_conversions.h"
+#include "tachyon/math/polynomials/univariate/radix2_evaluation_domain.h"
+#include "tachyon/math/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon::crypto {
+
+template <typename G1PointTy, typename G2PointTy>
+class KZGParams {
+ public:
+  using Fr = typename G1PointTy::ScalarField;
+  using Bucket = typename math::Pippenger<G1PointTy>::Bucket;
+  using UnivariateEvaluationDomainTy = math::Radix2EvaluationDomain<Fr>;
+  using DenseCoeffs = typename UnivariateEvaluationDomainTy::DenseCoeffs;
+  using G1JacobianPoint = typename G1PointTy::JacobianPointTy;
+
+  KZGParams() = default;
+
+  KZGParams(size_t k, std::vector<G1PointTy>&& g1_powers_of_tau,
+            std::vector<G1PointTy>&& g1_powers_of_tau_lagrange,
+            G2PointTy tau_g2)
+      : k_(k),
+        n_(1 << k),
+        g1_powers_of_tau_(std::move(g1_powers_of_tau)),
+        g1_powers_of_tau_lagrange_(std::move(g1_powers_of_tau_lagrange)),
+        tau_g2_(tau_g2) {}
+
+  size_t GetDegree() const { return k_; }
+
+  size_t GetSize() const { return n_; }
```

Comment:
Is there a reason for naming it as `GetDegree()` and `GetSize()`?

```suggestion
  size_t k() const { return k_; }

  size_t n() const { return n_; }
```